keps/sig-storage/2605-option-to-preserve-default-file-mode-bit-in-atomicwriter-volumes/README.md

```diff
@@ -0,0 +1,170 @@
+# Option to Preserve default file mode bit (Mode\DefaultMode) set in AtomicWriter volumes
+
+## Table of Contents
+
+<!-- toc -->
+- [Release Signoff Checklist](#release-signoff-checklist)
+- [Summary](#summary)
+- [Motivation](#motivation)
+- [Proposal](#proposal)
+  - [Preserve DefaultMode Flag](#preserve-defaultmode-flag)
+  - [File Permission](#file-permission)
+    - [Proposed heuristics](#proposed-heuristics)
+    - [Alternatives considered](#alternatives-considered)
+  - [Scope Of the Change](#scope-of-the-change)
+- [Graduation Criteria](#graduation-criteria)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+- [x] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+## Summary
+
+Kubernetes allows users to use Secret (and other configs) as a VolumeSource
+in pod spec. It also allows users to set a default file mode at the secret
+or every file level.
+The security context set for the pod (runAsUser or fsGroup) conflicts with
+default mode specified for the secret\file. In current implementation of the
+AtomicWriter volumes, default file mode is overwritten by the runAsUser or fsGroup
+bit while setting the volume ownership.
+
+Above results in unexpected behavior in cases where users want to run multiple
+containers in a single pod and want them to run under a fsGroup, have different UIDs etc.
+and yet have the ability to share same secret\config volume across containers.
```

Comment:
Maybe mention one example: a user might want to mount private ssh key secret with permission 0400. Or some git operation requires strict file permission which cannot be satisfied when the file has group permission.

Reply:
Some other known issue would be for cases like istio sidecar injection, it will automatically apply a fsGroup which can break the default Mode setting.

Reply:
added one example.